plugin/pkg/scheduler/scheduler_test.go

```diff
@@ -232,7 +232,9 @@ func TestSchedulerForgetAssumedPodAfterDelete(t *testing.T) {
 			return nil
 		}},
 		NextPod: func() *api.Pod {
-			return queuedPodStore.Pop().(*api.Pod)
+			var pod *api.Pod
+			queuedPodStore.Pop(func(obj interface{}) error { pod = obj.(*api.Pod); return nil })
```

Comment:
What we can potentially do is to have both "Pop()" and "PodAndRunUnderLock(func)" methods.
However, this is risky, because there people will be using the former and we can face the same races again in the future.

What we can do is to have a helper function that would be doing roughly this. WDYT?